daemon/volumes.go

```diff
@@ -205,11 +236,22 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 		}
 
 		if len(bind.Name) > 0 && len(bind.Driver) > 0 {
+			// set the label
+			if err := label.SetFileCreateLabel(container.MountLabel); err != nil {
+				return fmt.Errorf("Unable to setup default labeling for volume creation %s: %v", bind.Source, err)
+			}
+
+			// create the volume
 			v, err := createVolume(bind.Name, bind.Driver)
 			if err != nil {
 				return err
```

Comment:
How bad would it be to return on this error case without the label being reset? (cc @rhatdan)